### PR TITLE
Fixed Deviator auto targetting not prioritizing vehicles

### PR DESCRIPTION
**Pull request by Mailaender**
This makes them slightly more useful and bots a bit smarter.

### CODE REVIEW

**Comment by pchote** [comment]
What are the conceptual concerns here? The deviator doesn't affect non-vehicles, so shouldn't this be considered the same as 817e6f1ec16fdb37e648a128bc93ff65f62f948b etc?

**Comment by Porenutak** [comment]
Concerns is that its OK to have **AutoTarget** used by AI but not by players. **It should be player skill to aim correct  targets.** And its not the same as https://github.com/OpenRA/OpenRA/commit/817e6f1ec16fdb37e648a128bc93ff65f62f948b as Deviator do some damage to all units (small among of damage, but its still a damage.)
Please use Autotarget just when AI is in control, for example like this:

AutoTargetPriority@DeviatorTargetingUsedByBots:
		ValidTargets: Vehicle
		InvalidTargets: Infantry, Structure, Defense
		Priority: 50
		RequiresCondition: bot-owned && (attack-move || assault-move)

File: mods/d2k/rules/defaults.yaml [hunk 1]
```diff
@@ -153,6 +153,29 @@
 		AttackMoveCondition: attack-move
 		AssaultMoveCondition: assault-move
 
+^AutoTargetVehicleAssaultMove:
+	Inherits: ^AutoTargetGround
```
Comment (abcdefg30) [inline mods/d2k/rules/defaults.yaml]: Can't this inherit `^AutoTargetGroundAssaultMove`?

**Review by abcdefg30 — COMMENTED** [review]
Can we please also fix the names of the first two commits? They should be something along the lines of
- `Removed the water terrain type from Dune 2000.`
- `Fixed the description of ChangeOwnerWarhead`

File: mods/d2k/rules/defaults.yaml [hunk 2]
```diff
@@ -153,16 +153,24 @@
 		AttackMoveCondition: attack-move
 		AssaultMoveCondition: assault-move
 
+^AutoTargetVehicleAssaultMove:
+	Inherits: ^AutoTargetGroundAssaultMove
+	AutoTargetPriority@VEHICLES:
+		ValidTargets: Vehicle
+		InvalidTargets: Infantry, Structure, Defense
+		RequiresCondition: bot-owned && (attack-move || assault-move)
```
Comment (PunkPun) [inline mods/d2k/rules/defaults.yaml]:
```suggestion
		RequiresCondition: bot-owned
```

Shouldn't this priority be universally applied? If this priority doesn't return any targets it should move onto the next priority correct?

File: mods/d2k/rules/defaults.yaml [hunk 3]
```diff
@@ -153,16 +153,24 @@
...
+		Priority: 50
```
Comment (PunkPun) [inline mods/d2k/rules/defaults.yaml]: Is there any specific reason why it's 50?

I just want to know the precedence. In build queue we define in intervals of 10 IIRC

File: mods/d2k/rules/defaults.yaml (hunk 2 above)
Comment (Porenutak) [reply mods/d2k/rules/defaults.yaml]: `bot-owned && (attack-move || assault-move)` is used elsewhere in yaml. I was trying to stay with convention.

File: mods/d2k/rules/defaults.yaml (hunk 3 above)
Comment (Porenutak) [reply mods/d2k/rules/defaults.yaml]: > Is there any specific reason why it's 50?

no

File: mods/d2k/rules/defaults.yaml (hunk 2 above)
Comment (PunkPun) [reply mods/d2k/rules/defaults.yaml]: It's not used everywhere. Only where it is needed

Comment (Mailaender) [reply mods/d2k/rules/defaults.yaml]: I think this is a leftover from my initial attempt where I made player controlled Deviators smart, too.

**Review by PunkPun — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5f171cf0119bddb69e06bc70724da17960480eb1...106043a052483affac82961271c94058e12c8490)

**Pull request merged**